ubuntu_image/__main__.py

```diff
@@ -283,18 +285,34 @@ def parseargs(argv=None):
         default=None, action='append',
         help=_("""Extra ppas to install. This is passed through to
         livecd-rootfs."""))
+    '''
+    One option to support seeding in clasic if we add an argument
+    for classic subcommand to specify seeds dir. i.e:
+    $ ubuntu_image classic --seed-dir seeding gadget_tree
+
+    classic_cmd.add_argument(
+        '--seeding-dir',
+        default=None, metavar='SEEDING-DIR',
+        help=_("""A directory holds a variety of snaps for seeding in classic
+        image."""))
+    '''
```

Comment:
I am not quite sure if we're able to use hook API to do the similar stuff. I'll take a look at the hook API test case as a reference.

Reply:
The hook API should be able to do what you want. SInce from the code I saw the only thing you are doing is doing a copytree() operation from one directory into the rootfs - and that's exactly what you can do with the post-populate-rootfs hook. It's purpose is to customize the rootfs after creation. So you can just add a hook that copies the seed_dir into the rootfs as needed.